Assertion error when running `basilisp test` wihtout speicfying a path

Hi

there appears to be an general issue with Basilisp 0.3.6 when running `basilisp test` without explicitly specifying a path:
```
AssertionError: Must have at least one module name
```

To reproduce, 
1. create a new Poetry Basilisp project, and create a `tests/issue_test.lpy` file with a minimal test case (or clone https://github.com/ikappaki/issue-bas-test-path-dot)
```clojure 
(ns tests.issue-test
  (:require [basilisp.test :refer [deftest is]]))

(deftest abc
  (is (= 5 0)))
```
2. Run `basilisp test` the above error is returned:
``` shell
$ poetry run basilisp test
=================================================== test session starts ===================================================
platform win32 -- Python 3.11.4, pytest-8.3.4, pluggy-1.5.0
rootdir: d:\bas\issue-bas-test-path-dot
configfile: pyproject.toml
plugins: basilisp-0.3.6
collected 0 items / 1 error

__________________________________________ ERROR collecting tests/issue_test.lpy __________________________________________
.venv\Lib\site-packages\basilisp\contrib\pytest\testrunner.py:270: in collect
    module = self._import_module()
.venv\Lib\site-packages\basilisp\contrib\pytest\testrunner.py:247: in _import_module
    assert modnames, "Must have at least one module name"
E   AssertionError: Must have at least one module name
E   assert []
...
```

A workaround is to include the FQD name of the project directory when running `basilisp test`:
```shell
$ poetry run basilisp test -p <project dir>
platform win32 -- Python 3.11.4, pytest-8.3.4, pluggy-1.5.0
rootdir: d:\bas\issue-bas-test-path-dot
configfile: pyproject.toml
plugins: basilisp-0.3.6
collected 1 item

tests\issue_test.lpy F                                                                                               [100%]

___________________________________________________________ abc ___________________________________________________________
FAIL in (abc) (issue_test.lpy:5)
    Test failure: (= 5 0)

    expected: 5
      actual: 0
...
```

This issue seems to be related to a regression in (#1176), which I thought I previously tested with success, but it was in a slightly more complex setup that did not cover this basic case.

PR to follow.

Thanks

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed 
+ * Fixed a regression introduced in #1176 where the testrunner couldn't handle relative paths in `sys.path`, causing `basilisp test` to fail when no arugments were provided (#1204)
+
 ## [v0.3.6]
 ### Added
  * Added support for the `:decorators` meta key in anonymous `fn`s (#1178)

--- a/src/basilisp/cli.py
+++ b/src/basilisp/cli.py
@@ -745,7 +745,7 @@ def test(
             "Cannot run tests without dependency PyTest. Please install PyTest and try again.",
         )
     else:
-        pytest.main(args=list(extra))
+        sys.exit(pytest.main(args=list(extra)))
 
 
 @_subcommand(
@@ -768,7 +768,9 @@ def test(
         If all options are unambiguous (e.g. they are only either used by Basilisp
         or by PyTest), then you can omit the `--`:
 
-            `basilisp test -k vector -p other_dir`"""
+            `basilisp test -k vector -p other_dir`
+
+        Returns the PyTest exit code as the exit code."""
     ),
     handler=test,
     allows_extra=True,

--- a/src/basilisp/contrib/pytest/testrunner.py
+++ b/src/basilisp/contrib/pytest/testrunner.py
@@ -17,6 +17,7 @@ from basilisp.lang import runtime as runtime
 from basilisp.lang import symbol as sym
 from basilisp.lang import vector as vec
 from basilisp.lang.obj import lrepr
+from basilisp.lang.util import munge
 from basilisp.util import Maybe
 
 _EACH_FIXTURES_META_KW = kw.keyword("each-fixtures", "basilisp.test")
@@ -183,9 +184,10 @@ def _get_fully_qualified_module_names(file: Path) -> list[str]:
     there, we derive a Python module name referring to the given module path."""
     paths = []
     for pth in sys.path:
-        root = Path(pth)
+        root = Path(pth).resolve()
         if file.is_relative_to(root):
-            elems = list(file.with_suffix("").relative_to(pth).parts)
+            elems = list(file.with_suffix("").relative_to(root).parts)
+
             if elems[-1] == "__init__":
                 elems.pop()
             paths.append(".".join(elems))
@@ -269,6 +271,12 @@ class BasilispFile(pytest.File):
         filename = self.path.name
         module = self._import_module()
         ns = module.__basilisp_namespace__
+
+        # Ensure the test module was loaded because it was directly
+        # relative to an entry in `sys.path`.
+        if module.__name__ != munge(str(ns)):
+            raise ModuleNotFoundError(f"Module named '{ns}' is not in sys.path")
+
         once_fixtures, each_fixtures = self._collected_fixtures(ns)
         self._fixture_manager = FixtureManager(once_fixtures)
         for test in self._collected_tests(ns):

--- a/tests/basilisp/testrunner_test.py
+++ b/tests/basilisp/testrunner_test.py
@@ -284,11 +284,6 @@ def test_basilisp_test_noargs(pytester: pytest.Pytester):
     cmd = [basilisp, "test"]
     result = subprocess.run(cmd, capture_output=True, text=True, cwd=pytester.path)
 
-    print(f"\n\n--cmd--start: {' '.join(cmd)}")
-    print(f"\n\n--stdout--\n\n {result.stdout.strip()}")
-    print(f"\n\n--stderr--:\n\n {result.stderr.strip()}")
-    print(f"\n\n--cmd--end--: {' '.join(cmd)}\n\n")
-
     assert "==== 1 passed" in result.stdout.strip()
 
     assert result.returncode == 0

--- a/tests/basilisp/testrunner_test.py
+++ b/tests/basilisp/testrunner_test.py
@@ -1,4 +1,6 @@
 import platform
+import shutil
+import subprocess
 import sys
 
 import pytest
@@ -263,6 +265,35 @@ def test_fixtures_with_errors(
     result.assert_outcomes(passed=passes, failed=failures, errors=errors)
 
 
+def test_basilisp_test_noargs(pytester: pytest.Pytester):
+    runtime.Namespace.remove(sym.symbol("a.test-path"))
+
+    code = """
+    (ns tests.test-path
+      (:require
+       [basilisp.test :refer [deftest is]]))
+    (deftest passing-test
+      (is true))
+    """
+    pytester.makefile(".lpy", **{"./tests/test_path": code})
+
+    # I couldn't find a way to directly manipulate the pytester's
+    # `sys.path` with the precise control needed by this test, so we're
+    # invoking `basilisp test` directly as a subprocess instead ...
+    basilisp = shutil.which("basilisp")
+    cmd = [basilisp, "test"]
+    result = subprocess.run(cmd, capture_output=True, text=True, cwd=pytester.path)
+
+    print(f"\n\n--cmd--start: {' '.join(cmd)}")
+    print(f"\n\n--stdout--\n\n {result.stdout.strip()}")
+    print(f"\n\n--stderr--:\n\n {result.stderr.strip()}")
+    print(f"\n\n--cmd--end--: {' '.join(cmd)}\n\n")
+
+    assert "==== 1 passed" in result.stdout.strip()
+
+    assert result.returncode == 0
+
+
 def test_ns_in_syspath(pytester: pytest.Pytester, monkeypatch: pytest.MonkeyPatch):
     runtime.Namespace.remove(sym.symbol("a.test-path"))
 
@@ -324,11 +355,18 @@ def test_ns_not_in_syspath(pytester: pytest.Pytester):
       (:require
        [basilisp.test :refer [deftest is]]))
     """
-    pytester.makefile(".lpy", **{"./test/a/test_path": code})
+    # In this test, we use a `testabc` directory instead of `test`, as
+    # the latter can cause issues on macOS.  Specifically, macOS has a
+    # `/Library/Frameworks/Python.framework/Versions/3.xx/lib/python3.13/test`
+    # directory is picked up, resulting in a slightly different error
+    # message.
+    pytester.makefile(".lpy", **{"./testabc/a/test_path": code})
     pytester.syspathinsert()
-    result: pytest.RunResult = pytester.runpytest("test")
+    result: pytest.RunResult = pytester.runpytest("testabc")
     assert result.ret != 0
-    result.stdout.fnmatch_lines(["*ModuleNotFoundError: No module named 'test.a'"])
+    result.stdout.fnmatch_lines(
+        ["*ModuleNotFoundError: Module named 'a.test-path' is not in sys.path"]
+    )
 
 
 def test_ns_with_underscore(pytester: pytest.Pytester):

